src/HearThis/Script/ScriptProviderBase.cs

```diff
@@ -84,38 +85,51 @@ protected void Initialize()
 				throw new InvalidOperationException("Initialize should only be called once!");
 
 			LoadSkipInfo();
+			LoadProjectSettings();
 			DoDataMigration();
 		}
 
-		private void DoDataMigration()
+		private void LoadProjectSettings()
 		{
-			_dataVersionfilePath = Path.Combine(ProjectFolderPath, kProjectInfoFilename);
-			ProjectInfo projInfo;
-			if (File.Exists(_dataVersionfilePath))
-			{
-				projInfo = XmlSerializationHelper.DeserializeFromFile<ProjectInfo>(_dataVersionfilePath);
-			}
+			_projectSettingsFilePath = Path.Combine(ProjectFolderPath, kProjectInfoFilename);
+			if (File.Exists(_projectSettingsFilePath))
+				_projectSettings = XmlSerializationHelper.DeserializeFromFile<ProjectSettings>(_projectSettingsFilePath);
 			else
+				_projectSettings = new ProjectSettings();
+		}
+
+		public void SaveProjectSettings()
+		{
+			if (_projectSettings == null)
+				throw new InvalidOperationException("Initialize must be called first.");
+			XmlSerializationHelper.SerializeToFile(_projectSettingsFilePath, _projectSettings);
+		}
+
+		private void DoDataMigration()
+		{
+			while (_projectSettings.Version < Settings.Default.CurrentDataVersion)
 			{
-				// The first data migration is to go from "nothing" to version 1
-				projInfo = new ProjectInfo();
-			}
-			while (projInfo.Version < Settings.Default.CurrentDataVersion)
-			{
-				switch (projInfo.Version)
+				switch (_projectSettings.Version)
 				{
 					case 0:
 						//This corrects data corrupted by
 						// having recorded clips for blocks marked with a skipped style.
 						foreach (var style in _skippedParagraphStyles)
 							BackUpAnyClipsForSkippedStyle(style);
+						goto case 1;
```

Comment:
Looks like we're using two conflicting strategies for getting up to the desired version number. We should either have a while loop that gradually increments the version number taking it through each case statement OR require each case statement to have a "goto" to get it to the next version number. As it is, it looks to me as if it will come into case 0, do the migration for 0 and then go to 1 and do that migration. At that point, it will increment the version to 1 and come back through the switch statement and hit case 1 again, unnecessarily repeating that migration.